traffic_ops/app/lib/API/DeliveryServiceRegexes.pm

```diff
@@ -222,6 +262,18 @@ sub delete {
 		return $self->forbidden();
 	}
 
+	my $ds = $self->db->resultset('Deliveryservice')->find( { id => $ds_id } );
+	if ( !defined($ds) ) {
+		#allow deletion if the ds is not valid
```

Comment:
i don't think you can have an entry in deliveryservice_regex without the same delivery service in the deliveryservice table due to the foreign key in deliveryservice_regex. 

if that happens then you have a data integrity problem and someone will have to clean up the database manually and whatever bug is allowing this to happen needs to be fixed. So i would just stick with the basic:

my $ds = $self->db->resultset('Deliveryservice')->find( { id => $ds_id } );
	if ( !defined($ds) ) {
		return $self->not_found();
	}